juju/controller.py

```diff
@@ -437,6 +437,28 @@ def kill(self):
         """
         raise NotImplementedError()
 
+    async def cloud(self, name=None):
+        """Get Cloud
+
+        :param str name: Cloud name. If not specified, the cloud where
+                         the controller lives on is returned.
+        :returns: -> ~CloudResult
+        """
+        if name is None:
+            name = await self.get_cloud()
+        entity = client.Entity(tag.cloud(name))
+        cloud_facade = client.CloudFacade.from_connection(self.connection())
+        cloud = await cloud_facade.Cloud(entities=[entity])
+        return cloud.results[0]
```

Comment:
We should check that the results length is valid, before trying to access the results. That way we can give better error messages to the user.